### ISSUE TITLE
QPoint not supportint point arrays

### ISSUE THREAD

**Issue by dsoronda** [dsoronda/QtSharpDemos#4]
@ddobrev, can you take a look on this?
I'm guessing this should work?

``` csharp

            var points = new List<QPoint>();
            points.Add ( new QPoint ( 130, 140 ) );
            points.Add ( new QPoint ( 180, 170 ) );
            points.Add ( new QPoint ( 180, 140 ) );
            points.Add ( new QPoint ( 220, 110 ) );
            points.Add ( new QPoint ( 140, 100 ) );

            // http://doc.qt.io/qt-5/qpolygon.html#QPolygon-2 , we should be able to pase point list
            var polygon = new QPolygon(points);
            painter.DrawPolygon ( polygon );
```

**Comment by ddobrev** [comment]
Yes, it should, it's a bug. You cannot use Polygon because you cannot construct it, the relevant constructor is missing because it's a template type and I don't support those yet. Passing an array of points is not supported either because of the ambiguity C++ has about any pointer - is it a single object or an array?
I am busy again but this time I'll be able to take a look in just two weeks. Meanwhile please remove the last remaining comments about the now fixed crash at https://github.com/dsoronda/QtSharpDemos/blob/master/source/QtSharpDemos/GuiExample/PaintDemo.cs#L29 and https://github.com/dsoronda/QtSharpDemos/blob/master/source/QtSharpDemos/GuiExample/PaintShapesDemo.cs#L26 .
If you would like to keep writing examples while waiting, I have an idea, just let me know if you are interested.

**Comment by dsoronda** [comment]
@ddobrev 
I cleaned up those comments from code.
Also send em info about your idea, i'm always interested in new stuff.

**Comment by ddobrev** [comment]
Thank you.
My idea is that you could port the C++ Qt examples. They are really impressive and they come with each Qt installation so you already have their code. C++ is usually horrible to port but Qt's API in particular has always been simple and intuitive so I believe a large part of the examples would be easy to translate.

**Comment by ddobrev** [comment]
@dsoronda good news. I managed to work around this one for the time being. Use the following code:

```
var polygon = new QPolygon();
var points = new int[] { 130, 140, 180, 170, 180, 140, 220, 110, 140, 100 };
fixed (int* point = points)
{
    polygon.SetPoints(points.Length / 2, ref *point);
}
painter.DrawPolygon(polygon);
```

**Comment by ddobrev** [comment]
@dsoronda I can see at https://github.com/dsoronda/QtSharpDemos/blob/master/source/QtSharpDemos/GuiExample/Paint_ShapesDemo.cs#L57 that you haven't changed the example, is there a problem with this code?

**Comment by dsoronda** [comment]
@ddobrev 
Sorry, i had to take a break few days.
Example works but it is unsafe code. So I'll push changes but leave issue open.

**Comment by ddobrev** [comment]
I agree, it's a workaround anyway.
About your break, no problem and certainly no need to apologise. :) I just wanted to ensure you are not blocked here.
